src/sagemaker/jumpstart/curated_hub/jumpstart_curated_public_hub.py

```diff
@@ -0,0 +1,362 @@
+            if ex.response["Error"]["Code"] != "ResourceNotFound":
+                raise ex
+            return True
+
+    def _import_models(self, model_specs: List[JumpStartModelSpecs]):
+        print(f"Importing {len(model_specs)} models to curated private hub...")
+        tasks = []
+        with futures.ThreadPoolExecutor(
+            max_workers=self._default_thread_pool_size,
+            thread_name_prefix="import-models-to-curated-hub",
+        ) as deploy_executor:
+            for model_spec in model_specs:
+                task = deploy_executor.submit(self._delete_and_import_model, model_spec)
+                tasks.append(task)
+
+        results = futures.wait(tasks)
+        failed_deployments: List[BaseException] = []
+        for result in results.done:
+            exception = result.exception()
+            if exception:
+                failed_deployments.append(exception)
+        if failed_deployments:
+            raise RuntimeError(
+                f"Failures when importing models to curated hub in parallel: {failed_deployments}"
+            )
+
+    def _delete_and_import_model(self, model_spec: JumpStartModelSpecs):
+        self._delete_model_from_curated_hub(
+            model_spec
+        )  # TODO: Figure out why Studio terminal is passing in tags to import call
+        self._import_model(model_spec)
+
+    def _import_model(self, public_js_model_specs: JumpStartModelSpecs) -> None:
+        print(
+            f"Importing model {public_js_model_specs.model_id}"
+            + " version {public_js_model_specs.version} to curated private hub..."
+        )
+        self._content_copier.copy_hub_content_dependencies_to_hub_bucket(
+            model_specs=public_js_model_specs
+        )
+        self._import_public_model_to_hub(model_specs=public_js_model_specs)
+        print(
+            f"Importing model {public_js_model_specs.model_id}"
+            + " version {public_js_model_specs.version} to curated private hub complete!"
+        )
+
+    def _import_public_model_to_hub(self, model_specs: JumpStartModelSpecs):
+        # TODO Several fields are not present in SDK specs as they are only in Studio specs right now (not urgent)
+        hub_content_display_name = self.studio_metadata_map[model_specs.model_id]["name"]
+        # TODO enable: self.studio_metadata_map[model_specs.model_id]["desc"]
+        hub_content_description = "This is a curated model based "
+        +f"off the public JumpStart model {hub_content_display_name}"
+        hub_content_markdown = self._dst_s3_filesystem.get_markdown_s3_reference(
+            model_specs
+        ).get_uri()
+
+        hub_content_document = self._document_creator.make_hub_content_document(
+            model_specs=model_specs
+        )
+
+        self._sm_client.import_hub_content(
+            HubName=self.curated_hub_name,
+            HubContentName=model_specs.model_id,
+            HubContentVersion=model_specs.version,
```

Comment:
will `model_specs.version` ever be `*`?

Reply:
Not at this line of code - it converts all model specs to have the version at the start of the code path